localstack/utils/collections.py

```diff
@@ -104,7 +104,7 @@ class PaginatedList(List[_ListType]):
 
     def get_page(
         self,
-        token_generator: Callable[[_ListType], str],
+        token_generator: Optional[Callable[[_ListType], str]] = None,
```

Comment:
Since the index-based token generator is not how AWS services usually handle this, I think it's dangerous to use it by default (since it encourages us to move further away from AWS in terms of parity).